### DISCUSSION TITLE
AIdeck & crazyflie launch.py for using yolo

### DISCUSSION THREAD

**Discussion by wyseo10** [bitcraze/discussions#1660] [Q&A]
Hello! I am currently working on combining real-time object detection using YOLOv8 with Crazyflie flight by utilizing the AI Deck and Flow Deck.

However, I have encountered an issue where I cannot run both the process of retrieving raw files from the AI Deck in real time and the Crazyflie launch.py script simultaneously.

Is it fundamentally challenging to execute both tasks in real time?
Additionally, if there are any similar discussions or related resources, could you kindly share them with me?

Thank you in advance!

**Comment by gemenerik** [comment]
Hi! Exciting that you are working on running YOLO on the AI deck. I'm not sure what the `launch.py` script is, can you tell me a bit more about what it is and what it does? Are you trying to control the Crazyflie over the radio, and stream images from the AI-deck over WiFi? It could be that the radio is interfering with the WiFi connection. Maybe you can try some different radio channels and see if that improves things.

**Reply by wyseo10** [reply]
Hi! Thank you for your response.

The launch.py script I’m using is from [this repository](https://github.com/IMRCLab/crazyswarm2/blob/main/crazyflie/launch/launch.py). Its purpose is to check if the communication with the Crazyflie is working properly and to monitor the Crazyflie's battery status.

Currently, I am using Crazyradio 2.0 to communicate with the Crazyflie, and I am streaming images from the AI Deck via WiFi. For the image streaming, I am using [this example code](https://github.com/bitcraze/aideck-gap8-examples/tree/master/examples/other/wifi-img-streamer) as a base.

I have already tried different radio channels (10, 20, 40, and 80), but unfortunately, it still doesn’t work. Do you have any additional suggestions?

**Reply by gemenerik** [reply]
That's a bummer. Another you can try is reducing the tx power on the Crazyflie side.
You can call [this function](https://github.com/bitcraze/crazyflie-firmware/blob/7c64c98060db7622c77c808d97797abb59ef9e9f/src/hal/src/radiolink.c#L143) from the STM firmware to adjust it.
